datalad/support/gitrepo.py

```diff
@@ -1296,14 +1320,16 @@ def commit(self, msg=None, options=None, _datalad_msg=False, careless=True,
         # Therefore, for now always use direct call.
         # TODO: Figure out, what exactly is going on with gitpython here
 
-        cmd = ['git', 'commit'] + (["-m", msg if msg else ""])
+        cmd = ['git', 'commit']
         if options:
             cmd.extend(options)
         lgr.debug("Committing via direct call of git: %s" % cmd)
 
         try:
             self._git_custom_command(files, cmd,
-                                     expect_stderr=True, expect_fail=True,
+                                     log_stdout=not allow_interactive,
+                                     expect_stderr=True,
+                                     expect_fail=True,
                                      check_fake_dates=True,
                                      index_file=index_file)
         except CommandError as e:
```

Comment:
```
add(ok): /tmp/datalad_demo_bob.0wlc/myanalysis/code (directory)
[ERROR  ] argument of type 'NoneType' is not iterable [gitrepo.py:commit:1336] (TypeError) 
I: done running docs/examples/3rdparty_analysis_workflow.sh: failed
The command "if [ ${DATALAD_TESTS_SSH:-0} != 1 -o -n "${BUILD_DATALAD_EXTENSION:-}" ] || echo "${TMPDIR:-}" | grep -q ' '; then echo "Testing examples is disabled by configuration"; else $NOSE_WRAPPER ../tools/testing/run_doc_examples; fi" exited with 1.
```
With this change stdout could be passed through and would be `None` here.

Reply:
Not sure what you mean. Will research it in the next iteration.

Reply:
if log_stdout is False, then e.stdout is None, and thus this crash

That is the kind of "CI failures" I was talking about where previous we might just commit with empty commit message when after this particular change we would end up either crashing like here or hanging asking "CI users" to enter commit message

Reply:
Hmm, I think the fact that it crashes is exactly the desired behavior, because the code (and myself), are unaware of these inter-dependencies. The changes aim to not affect CI behavior in any way.

Reply:
Looked into it, the whole thing seems flawed. If I let stdout through to the caller, error messages are not properly extracted, and things like `Repo.commit(careless=True)` are not functional (and that is a critical issue for us). So indeed, and as feared, this is not a CI issue, but a more global problem. I could not yet determine, whether a particular combination of Runner options would make it possible to act on errors and have stdout be connected to the calling terminal. The numerical error codes seem unspecific (e.g. 'nothing to commit' has `1`, like all the others).

If there is no solution to this, we can scrap the whole thing and stay with `[DATALAD] Recorded changes` :(